### PR TITLE
Tmnt

### PR DESCRIPTION
**Pull request by trietmnj**
Css linking is  still messed up somewhere, the login banner looks weird. Functional though.

### CODE REVIEW

**Comment by trietmnj** [comment]
This pull request implements authentication and authentication lockout from the main map page

File: src/containers/main/main.js [hunk 1]
```diff
@@ -3,16 +3,25 @@ import { connect } from 'redux-bundler-react';
...
-        <Map/>
-      </div>
-    )
+function MainPage(props) {
+
+  const {doUpdateUrl,authNSIToken} = props;
+
+  // not logged not
+  if(!authNSIToken) {
+    doUpdateUrl("/");
```
Comment (HenryGeorgist) [inline src/containers/main/main.js:12]: looks great.

File: .env.development [hunk 2]
```diff
@@ -1,2 +1,4 @@
 HTTPS=true
-BROWSER='chrome'
\ No newline at end of file
+BROWSER='chrome'
+REACT_APP_NSI_AUTH_HOST=https://ml-dev.sec.usace.army.mil/auth/v3/token
```
Comment (HenryGeorgist) [inline .env.development:3]: good.

File: src/app-bundles/auth-bundle.js [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+const parseJwt = (token) => {
+  try {
+    return JSON.parse(atob(token.split('.')[1]));
+  } catch (e) {
+    return null;
+  }
+};
+
+// A bundle can have:
```
Comment (HenryGeorgist) [inline src/app-bundles/auth-bundle.js]: nice.

**Review by HenryGeorgist — APPROVED** [review]
looks good

**Pull request merged**